src/pointcloud_to_laserscan_node.cpp

```diff
@@ -225,9 +243,80 @@ void PointCloudToLaserScanNode::cloudCallback(
     if (range < scan_msg->ranges[index]) {
       scan_msg->ranges[index] = range;
+      v_pointcloud_index.at(index) = pointcloud_index;
     }
   }
-  pub_->publish(std::move(scan_msg));
+
+  laserscan_pub_->publish(std::move(scan_msg));
+
+  pcl::PointCloud<pcl::PointXYZ> pcl_pointcloud;
+  sensor_msgs::PointCloud2ConstIterator<float> iter_x(*cloud_msg, "x"), iter_y(*cloud_msg, "y"),
+    iter_z(*cloud_msg, "z");
+  for (size_t i = 0; i < v_pointcloud_index.size(); ++i) {
+    if (v_pointcloud_index.at(i) != no_data) {
+      pcl::PointXYZ point;
+      point.x = *(iter_x + v_pointcloud_index.at(i));
+      point.y = *(iter_y + v_pointcloud_index.at(i));
+      point.z = *(iter_z + v_pointcloud_index.at(i));
+      pcl_pointcloud.push_back(point);
+    }
+  }
+  pcl::toROSMsg(pcl_pointcloud, pointcloud_output);
+  auto pointcloud_output_ptr
+    = std::make_unique<sensor_msgs::msg::PointCloud2>(pointcloud_output);
+  pointcloud_output_ptr->header = cloud_msg->header;
+  pointcloud_pub_->publish(std::move(pointcloud_output_ptr));
+
+  // ray msg
+  ray_output.header = cloud_msg->header;
+  ray_output.id = 0;
+  ray_output.type = visualization_msgs::msg::Marker::LINE_LIST;
+  ray_output.action = visualization_msgs::msg::Marker::MODIFY;
+  ray_output.lifetime = rclcpp::Duration::from_seconds(0);
+  ray_output.scale.x = 0.05;
+  ray_output.color.a = 0.1;  // Don't forget to set the alpha!
+  ray_output.color.r = 1.0;
+  ray_output.color.g = 1.0;
+  ray_output.color.b = 0.0;
+  for (size_t i = 0; i < pcl_pointcloud.size(); ++i) {
+    geometry_msgs::msg::Point point;
+    point.x = 0;
+    point.y = 0;
+    point.z = 0;
+    ray_output.points.push_back(point);
+    point.x = pcl_pointcloud.at(i).x;
+    point.y = pcl_pointcloud.at(i).y;
+    point.z = pcl_pointcloud.at(i).z;
+    ray_output.points.push_back(point);
+  }
+
+  ray_viz_pub_->publish(ray_output);
+
+  // stixel msg
+  for (size_t i = 0; i < pcl_pointcloud.size(); ++i) {
+    visualization_msgs::msg::Marker marker;
+    marker.header = cloud_msg->header;
+    marker.id = i;
+    marker.type = visualization_msgs::msg::Marker::CUBE;
+    marker.action = visualization_msgs::msg::Marker::MODIFY;
+    marker.lifetime = rclcpp::Duration::from_seconds(0.2);
+    marker.color.a = 0.999;  // Don't forget to set the alpha!
+    marker.color.r = 0.8;
+    marker.color.g = 0.8;
+    marker.color.b = 0.8;
+    marker.scale.x = 0.2;
+    marker.scale.y = 0.2;
+    marker.scale.z = (pcl_pointcloud.at(i).z - min_height_);
+    marker.pose.position.x = pcl_pointcloud.at(i).x;
+    marker.pose.position.y = pcl_pointcloud.at(i).y;
+    marker.pose.position.z = pcl_pointcloud.at(i).z - marker.scale.z * 0.5;
+    marker.pose.orientation.x = 0.0;
+    marker.pose.orientation.y = 0.0;
+    marker.pose.orientation.z = 0.0;
+    marker.pose.orientation.w = 1.0;
+    stixel_output.markers.push_back(marker);
+  }
+  stixel_viz_pub_->publish(stixel_output);
```

Comment:
IMO, it's better to move these marker-related lines to `unnamed namespace` or `member functions` or somewhere.

Reply:
Fixed at https://github.com/tier4/pointcloud_to_laserscan/pull/1/commits/e8e73626c8ff1c97512c4d71f5d0001b9fb4dc07